cardano-ledger-core/src/Cardano/Ledger/Serialization.hs

```diff
@@ -203,9 +203,10 @@ ratioToCBOR r =
 ratioFromCBOR :: (Bounded a, Integral a, FromCBOR a) => Decoder s (Ratio a)
 ratioFromCBOR = decodeFraction fromCBOR
 
+-- | Decode a non-negative rational with a bounded representation.
 rationalFromCBOR :: Decoder s Rational
 rationalFromCBOR = do
-  x <- decodeFraction decodeInt64
+  x <- decodeFraction decodeWord64
```

Comment:
I'd suggest the return type of the function to depict the values it can produce, i.e.:
```
rationalFromCBOR :: Decoder s (Ratio Word64)
```
otherwise it looks like it can decode any `Rational`.